stdlib/LibGit2/src/callbacks.jl

```diff
@@ -186,9 +186,9 @@ function authenticate_userpass(libgit2credptr::Ptr{Ptr{Cvoid}}, p::CredentialPay
     if p.use_git_helpers && (!revised || !isfilled(cred))
         git_cred = GitCredential(p.config, p.url)
 
-         # Use `deepcopy` to ensure shredding the `git_cred` does not shred the `cred`s copy
+         # Use `copy` to ensure shredding the `git_cred` does not shred the `cred`s copy
         cred.user = something(git_cred.username, "")
-        cred.pass = deepcopy(something(git_cred.password, ""))
+        cred.pass = copy(something(git_cred.password, ""))
```

Comment:
Hm.  Of course this is introducing a test failure because `copy("")` is not defined.  We also cannot move the copy inside the `something` because `copy(nothing)` is also not defined.  And it feels silly to use a `SecretBuffer()` as the default value because then we need to both `copy` it and `shred!` it, but I suppose that's the best solution.

Reply:
Alternative:
```julia
cred.pass = git_cred.password !== nothing ? copy(git.cred_password) : ""
```